syncopy/spikes/spike_psth.py

```diff
@@ -128,40 +152,64 @@ def spike_psth(data,
             interval = [0, np.max(trl_ends)]
     # explicit time window in seconds
     else:
-        array_parser(latency, lims=[0, np.inf], dims=(2,))
+        array_parser(latency, lims=[-np.inf, np.inf], dims=(2,))
         interval = latency
 
     # --- determine overall (all trials) histogram shape ---
 
-    # get average trial size for auto-binning
+    # get average trial size for auto-binning (no selection respected)
     av_trl_size = data.data.shape[0] / len(data.trials)
 
-    # TODO: respect time window (latency)
     if binsize in available_binsizes:
         nBins = available_binsizes[binsize](av_trl_size)
         bins = np.linspace(*interval, nBins)
     else:
+        scalar_parser(binsize, varname='binsize', lims=[0, np.inf])
         # include rightmost bin edge
         bins = np.arange(interval[0], interval[1] + binsize, binsize)
         nBins = len(bins)
-    print(interval, bins)
-
-    # get all channelX-unitY combinations with at least one event
-
-    # possible channel-unit indice combinations (it's not a meshgrid..)
-    combs = []
 
-    # the straightforward way would be: np.unique(data.data[:, 1:], axis=0)
-    # however this loads 66% the size of the total data into memory
-    for trial in data.trials:
-        # tuples allow for a set operation, dimord is fixed as parsed above!
-        combs.append(np.unique(trial[:, 1:], axis=0))
-
-    combs = np.unique(np.concatenate(combs), axis=0)
+    # it's a sequential loop to get an array of [chan, unit] indices
+    combs = get_chan_unit_combs(trials)
 
     # right away create the output labels for the channel axis
     chan_labels = [f'channel{i}_unit{j}' for i, j in combs]
 
     # now we have our global (single-trial, avg, std,..) histogram shape
     h_shape = (nBins, len(combs))
```

Comment:
this is not part of this change set, but `chan_labels` and `h_shape` are also unused